examples/mfa/mfacosigner/server.go

```diff
@@ -0,0 +1,248 @@
+package mfacosigner
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/go-webauthn/webauthn/protocol"
+	"github.com/go-webauthn/webauthn/webauthn"
+	"github.com/lestrrat-go/jwx/v2/jwa"
+	"github.com/openpubkey/openpubkey/examples/mfa/mfacosigner/jwks"
+	"github.com/openpubkey/openpubkey/pktoken"
+	"github.com/openpubkey/openpubkey/util"
+)
+
+type Server struct {
+	cosigner *MfaCosigner
+	jwksUri  string
+}
+
+func NewMfaCosignerHttpServer(serverUri, rpID, rpOrigin, RPDisplayName string) (*Server, error) {
+	server := &Server{}
+
+	// WebAuthn configuration
+	cfg := &webauthn.Config{
+		RPDisplayName: RPDisplayName,
+		RPID:          rpID,
+		RPOrigin:      rpOrigin,
+	}
+
+	// Generate the key pair for our cosigner
+	alg := jwa.ES256
+	signer, err := util.GenKeyPair(alg)
+	if err != nil {
+		return nil, err
+	}
+
+	jwksServer, kid, err := jwks.NewJwksServer(signer, alg)
+	if err != nil {
+		return nil, err
+	}
+	jwksHost := jwksServer.URI()
+	server.jwksUri = fmt.Sprintf("%s/.well-known/jwks.json", jwksHost)
+	issuer := rpOrigin
+
+	fmt.Println("JWKS hosted at", server.jwksUri)
+	server.cosigner, err = New(signer, alg, issuer, kid, cfg)
+	if err != nil {
+		return nil, err
+	}
+
+	mux := http.NewServeMux()
+	mux.Handle("/", http.FileServer(http.Dir("mfacosigner/static")))
+	mux.HandleFunc("/mfa-auth-init", server.initAuth)
+	mux.HandleFunc("/check-registration", server.checkIfRegistered)
+	mux.HandleFunc("/register/begin", server.beginRegistration)
+	mux.HandleFunc("/register/finish", server.finishRegistration)
+	mux.HandleFunc("/login/begin", server.beginLogin)
+	mux.HandleFunc("/login/finish", server.finishLogin)
+	mux.HandleFunc("/sign", server.signPkt)
```

Comment:
Agreed, originally the auth cosigner was built using the OpenPubkey paper as the spec, but it evolved beyond that. 

I created this issue to track this

https://github.com/openpubkey/openpubkey/issues/85

Reply:
I think this would be important to include in this PR. If you're creating something that people need to integrate against, it's important to tell them what they need to build for the integration. This could be accomplished with an interface.

Reply:
> This could be accomplished with an interface.

If they are implementing a cosigner in golang, they should use the provided client and server code in `client` and `/cosigner` rather than writing this own from scratch. However that only applies to the AuthCosigner.

Is the interface discussion relevant to `examples/mfa/mfacosigner/server.go`?